### PR TITLE
Update dependencies & improve build performance

### PR DESCRIPTION
**Pull request by nhajratw**
Just some minor updates.

FYI, upgrading gradle to 7.0.2 (the latest), caused test failures that I didn't know how to solve, so I just updated to the latest 6.x series.

### CODE REVIEW

**Review by marcphilipp — APPROVED** [review]

**Comment by marcphilipp** [comment]
@nhajratw Thanks! 👍

File: gradle.properties [hunk 1]
```diff
@@ -0,0 +1,3 @@
+org.gradle.caching=true
+org.gradle.parallel=true
+org.gradle.vfs.watch=true
```
Comment (szpak) [inline gradle.properties:3]: I'm not convinced if we should enable it at the project level. It's rather user environment specific decision.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=7faa7198769f872826c8ef4f1450f839ec27f0b4d5d1e51bade63667cbccd205
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.9-bin.zip
```
Comment (szpak) [inline gradle/wrapper/gradle-wrapper.properties:3]: Why the is `distributionSha256Sum` missing?

**Review by szpak — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 1 above)
Comment (marcphilipp) [reply gradle.properties:3]: It's enabled by default in Gradle 7 so I think that's ok.

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (marcphilipp) [reply gradle/wrapper/gradle-wrapper.properties:3]: The IDEA issue with it has been fixed in the meantime. I'll re-add it when upgrading to Gradle 7.

Comment (nhajratw) [reply gradle/wrapper/gradle-wrapper.properties:3]: I just ran `./gradlew wrapper --gradle-version=6.9` -- this was the result.

Comment (marcphilipp) [reply gradle/wrapper/gradle-wrapper.properties:3]: @szpak was referring to the additional `--gradle-distribution-sha256-sum` option (cf. https://docs.gradle.org/current/userguide/gradle_wrapper.html#configuring_checksum_verification)

Comment (nhajratw) [reply gradle/wrapper/gradle-wrapper.properties:3]: yes, i know -- I'm just saying that the wrapper command seems to have removed that line. Not sure if it's not required....

Anyway , per that link you posted, it seems like you could add a GitHub Action to validate the wrapper checksum. Might be a good idea.

Comment (szpak) [reply gradle/wrapper/gradle-wrapper.properties:3]: @nhajratw The following command is better:
```
./gradlew wrapper --gradle-version 7.0.2 --distribution-type bin --gradle-distribution-sha256-sum 0e46229820205440b48a5501122002842b82886e76af35f0f3a069243dca4b3c
```
(with checksum available [here](https://gradle.org/release-checksums/))

Comment (szpak) [reply gradle/wrapper/gradle-wrapper.properties:3]: > Anyway , per that link you posted, it seems like you could add a GitHub Action to validate the wrapper checksum. Might be a good idea.

We have one, so maybe it even did that :)
https://github.com/gradle-nexus/publish-plugin/actions/workflows/gradle-wrapper-validation.yml

Comment (szpak) [reply gradle/wrapper/gradle-wrapper.properties:3]: > @szpak was referring to the additional --gradle-distribution-sha256-sum option

I missed that your command Marc. Thanks.

Comment (marcphilipp) [reply gradle/wrapper/gradle-wrapper.properties:3]: This action validates the checksum of the wrapper JAR that is checked in in `gradle/wrapper`. The `distributionSha256Sum` is about validating the checksum of the Gradle distribution that is downloaded by the wrapper.

**Pull request merged**